### DISCUSSION TITLE
`heroku/builder:24` has been released with multi-architecture and ARM support!

### DISCUSSION THREAD

**Discussion by joshwlewis** [heroku/buildpacks#25] [Announcements]
We have just released our latest [CNB Builder](https://buildpacks.io/docs/for-platform-operators/concepts/builder/) ,[`heroku/builder:24`](https://github.com/heroku/cnb-builder-images)! It includes brand new support for `ARM64` application builds!

Give it a try by building your app like this:

```
cd /path/to/application
pack build my-app --builder heroku/builder:24
```

### New multi-architecture support

If you're using an ARM64 computer (such as an Apple Silicon Mac or an AWS Graviton instance), `pack build my-app heroku/builder:24` will produce a `linux/arm64` image. If you're using an AMD64 computer (such as an Intel Mac, most desktops and laptops, or an AWS EC2 instance), the same command will produce a `linux/amd64` image.

You may also specify which architecture you wish to build with via the `--platform tag`, like this (`pack` version `0.34.0` or newer required):

```
pack build my-app --platform linux/arm64 --builder heroku/builder:24
```

### Updated base image and reduced size

`heroku/builder:24` is based on Heroku-24, which in turn is based on Ubuntu:24.04. Therefore, a large number of Ubuntu packages have been updated.

Additionally, we adjusted which Ubuntu packages are installed to reduce the image size:

- We removed some less frequently used packages (such as Bazaar, Mercurial and Ghostscript)
- We removed some tools that are typically only required during the build (such as GCC, Make, Git
      and *system* Python) from the run image, but left them in the build image.

**Comment by nickhammond** [comment]
Nice work on reducing the image size! The buildpack built image for one of my apps is only a few mbs bigger than the native Docker one which is great.

**Comment by garry-t** [comment]
Actually it is not clear how to make work your ruby application after you build it via buildpacks

jemalloc.sh - not found
puma - not found 
bundle - not found

project.toml
```
[_]
schema-version = "0.2"
[com.heroku.buildpacks.deb-packages]
install = [
    # string version of a dependency to install
    "libtag1-dev",
    "libjemalloc2",
    "memcached",
    "nginx"
]
```
How these buildpacks actually works ? :)

**Reply by edmorley** [reply]
(Duplicate of https://github.com/heroku/buildpacks/discussions/6#discussioncomment-11098473)
